src/floating_point_numbers/restrict_within_range.md

```diff
@@ -0,0 +1,22 @@
+# Restrict within range
+
+When doing some calculations you might want to enforce that the result falls within a certain range.
+
+In Java 21 `java.lang.Math` got expanded with a clamp function for `double`, `float`, `clamp` and `long` in the form of: `clamp(value, min, max)`
+
+So if we were to do:
+`double result = Math.clamp(100*100, 2, 50)` the result would be 50.
+
+In older versions we'd have to use a function like the following to handle this:
+
+```java
+static <T extends Comparable<T>> T clamp(T value, T min, T max) {
```

Comment:
The "older code" might be a bit a bit confusing for a beginner. 

Possibly something like:

```
public int clamp(int value, int min, int max) {
  return Math.max(min, Math.min(value, max));
}
```

Or furher expand it to use if the if statements e.g. 

```java
public int clamp(int value, int min, int max) {
  if (value > max) {
    return max;
  } else if (value < min) {
    return min;
  } else {
    return value;
  }
}
```

Reply:
Good point, would you keep the original as an example of the generic version thereof, or only show the base expanded version?

Reply:
Don't think the generic version is needed at all.